gateway/crates/config/src/entity_caching.rs

```diff
@@ -1,10 +1,62 @@
-use std::time::Duration;
+use std::{path::PathBuf, time::Duration};
 
 #[derive(Debug, Default, serde::Deserialize, Clone, PartialEq)]
 pub struct EntityCachingConfig {
     pub enabled: Option<bool>,
 
+    #[serde(default)]
+    pub storage: EntityCachingStorage,
+
+    #[serde(default)]
+    pub redis: Option<EntityCachingRedisConfig>,
+
     /// The ttl to store cache entries with.  Defaults to 60s
     #[serde(deserialize_with = "duration_str::deserialize_option_duration", default)]
     pub ttl: Option<Duration>,
 }
+
+#[derive(Debug, Clone, Default, PartialEq, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+pub enum EntityCachingStorage {
+    #[default]
+    Memory,
+    Redis,
+}
+
+#[derive(Debug, Clone, PartialEq, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct EntityCachingRedisConfig {
```

Comment:
Because entity caching and rate limiting are different features and could very easily require different configurations.  I'd rather not couple them together and have to unpick later.

Reply:
well up to you. IMHO that arguments doesn't hold for TLS config though

Reply:
It doesn't, but I'm also not sure if I care about one additional 3 field struct

Reply:
Sure, but my point remains - things aren't going to collapse under the weight of those additional 20 or so lines

Reply:
Again up to you, I've approved. I just don't see why entity caching would require some specific TLS config that would never happen in rate limiting. So I just don't see how it's not just duplicated code that will need to be updated together if we ever touch it, and if we never touch it, I don't see what adding two of them brings. But I understand the argument that you don't want to bother now that it's written.

Reply:
> the argument that you don't want to bother now that it's written.

🎯